packages/repository/src/__tests__/unit/model/model.unit.ts

```diff
@@ -194,6 +200,102 @@ describe('model', () => {
     });
   });
 
+  it('includes navigational properties in JSON (strict-mode)', () => {
+    class Category extends Entity {
+      id: number;
+      products: Product[];
+
+      constructor(data: Partial<Category>) {
+        super(data);
+      }
+    }
+
+    class Product extends Entity {
+      id: number;
+      categoryId: number;
+
+      constructor(data: Partial<Product>) {
+        super(data);
+      }
+    }
+
+    Category.definition = new ModelDefinition('Category')
+      .addSetting('strict', true)
+      .addProperty('id', {type: 'number', id: true, required: true})
+      .addRelation(<HasManyDefinition>{
+        name: 'products',
+        type: RelationType.hasMany,
+        targetsMany: true,
+
+        source: Category,
+        keyFrom: 'id',
+
+        target: () => Product,
+        keyTo: 'categoryId',
+      });
```

Comment:
nitpick: IIRC, `{...} as HasManyDefinition` is a recommended style over `<HasManyDefinition>{...}`.

Reply:
Makes sense. From what I remember, when using `{...} as HasManyDefinition`, TypeScript is not able to offer correct suggestions (auto-completion) for members of the object being created :(